Source/Scene/processPbrMaterials.js

```diff
@@ -688,10 +688,18 @@ define([
             fragmentShader += '    specularIrradiance = mix(specularIrradiance, belowHorizonColor, smoothstep(aroundHorizon, farBelowHorizon, reflectionDotNadir) * inverseRoughness);\n';
             fragmentShader += '    specularIrradiance = mix(specularIrradiance, nadirColor, smoothstep(farBelowHorizon, 1.0, reflectionDotNadir) * inverseRoughness);\n';
 
+            // Luminance model from page 40 of http://silviojemma.com/public/papers/lighting/spherical-harmonic-lighting.pdf
+            // Angle between sun and zenith
+            fragmentShader += '    float S = acos(clamp(dot(normalize(czm_inverseViewRotation * l), normalize(v_positionWC * -1.0)), 0.001, 1.0));\n';
+            // Angle between zenith and current pixel
+            fragmentShader += '    float theta = acos(clamp(dot(normalize(czm_inverseViewRotation * n), normalize(v_positionWC * -1.0)), 0.001, 1.0));\n';
+            // Angle between sun and current pixel
+            fragmentShader += '    float gamma = acos(NdotL);\n';
```

Comment:
For `S`, `theta`, and `gamma`, you don't need to use `acos` just to call ` cos` on it again. For example, in the luminance below, there is `pow(cos(gamma), 2.0)` which is `pow(cos(acos(NdotL)), 2.0` which can be simplified to just `NdotL * NdotL`. You can do that for all three.